pages/_app.tsx

```diff
@@ -16,55 +16,24 @@ interface UserData {
 }
 
 export default function App({ Component, pageProps }: AppProps) {
-  const [userStatus, setUserStatus] = useState<boolean>(false);
   const [userData, setUserData] = useState<UserData | null>(null);
-  const [sharedSearch, setSharedSearch] = useState<string>("");
-  const [userFolder, setUserFolder] = useState({
-    name: "",
-    userName: "",
-    img: "",
-  });
+  const [login, setLogin] = useState<string | null>(null);
```

Comment:
```suggestion
  const [login, setLogin] = useState<string>("");
```
해당 state에 null이 들어오게 되는 경우가 아니라면 state의 기본값을 넣어주는 것이 좋습니다.

그리고 userData와 login이라는 state는 Header에서만 사용되는 부분이기에
Header에서 선언하고 사용하는 것이 좋겠네요!

App 컴포넌트에서 사용하게 되면 불필요한 렌더링이 발생할 수 있습니다.